thrust/thrust/iterator/counting_iterator.h

```diff
@@ -196,18 +205,68 @@ class _CCCL_DECLSPEC_EMPTY_BASES counting_iterator
       : super_t(x)
   {}
 
+  _CCCL_HOST_DEVICE explicit counting_iterator(Incrementable x, StrideHolder stride)
+      : super_t(x)
+      , StrideHolder(stride)
+  {}
+
   //! \cond
 
 private:
+  template <typename S = StrideHolder>
+  auto stride() const
+  {
+    return static_cast<const S&>(*this).value;
+  }
```

Comment:
with mdspan we have `__de_ice` which extracts the value from an integer or and integral-constant-like

Reply:
I cannot use `__de_ice(I)` because it requires `I` to be `__integral_constant_like_`, which requires `convertible_to<_Tp, decltype(_Tp::value)>`. But this conversion would mess with `counting_iterator::operator+, as described in the comment above :/